### PR TITLE
640/leaderboard view page

### LINKED ISSUE #640: Leaderboard View Page
### What are you wanting to be added?

The website must display a leaderboard view page at `/game/{game}`.
The website may may display a 404 page if the leaderboard cannot be retrieved by slug from the backend.
The leaderboard view page must display the name of the leaderboard as a header.
The leaderboard view page must display the leaderboard info blurb.
The leaderboard info may be hidden by default and require an info section to be expanded by the user.

### What might break?

nothing's going to break don't worry about it

### Relevant Issues

_No response_
```[tasklist]
### Tasks
- [ ] https://github.com/leaderboardsgg/leaderboard-site/issues/647
```

### LINKED ISSUE #647: Feature Request: Change url path for boards to games
### What are you wanting to be added?

Change board path to game and update any references

### What might break?

Nothing should break

### Relevant Issues

_No response_

### PR DESCRIPTION
**Pull request by Eein**
## What

- changes board/:slug to game/:slug
- shows game name as header
- shows toggle-able info dropdown showing Leaderboard.Info
- if api returns a 404, renders the nuxt 404 page - This must be
  implemented globally and is out of scope of this PR
- does not render an info button if no info blurb exists or is empty string (this can be updated 

TODO:
- [ ] testing

blocked by: https://github.com/leaderboardsgg/leaderboard-backend/pull/256

## Link to Issue

closes #640 
closes #647

### Steps for testing

1. Test with text in info field
2. Test with blank string in info field

### Images

image

image

### CODE REVIEW

File: pages/game/[slug].vue [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
+import { useRoute, createError } from '#imports'
 import Loader from 'blocks/Loader/Loader.vue'
 import { useGetLeaderboardBySlug } from '~/composables/api'
-
+import LeaderboardInfo from '~/components/blocks/LeaderboardInfo/LeaderboardInfo.vue'
 const route = useRoute()
 const leaderboardSlug = route.params.slug as string
 const leaderboardDetail = await useGetLeaderboardBySlug(leaderboardSlug)
+
+if (leaderboardDetail?.error?.status == 404) {
```
Comment (TheTedder) [inline pages/game/[slug].vue:10]: This should be === right?

**Review by TheTedder — COMMENTED** [review]
How hard would it be to add a test to make sure the page displays a 404 error when it can't fetch the leaderboard?

**Comment by Eein** [comment]
I can do that! Lemme knock it out in a bit

**Comment by TheTedder** [comment]
Or at least verify that it creates the error.

**Review by TheTedder — CHANGES_REQUESTED** [review]

File: pages/game/[slug].vue [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<script setup lang="ts">
+import { useRoute, createError } from '#imports'
+import Loader from 'blocks/Loader/Loader.vue'
+import { useGetLeaderboardBySlug } from '~/composables/api'
+import LeaderboardInfo from '~/components/blocks/LeaderboardInfo/LeaderboardInfo.vue'
+const route = useRoute()
+const leaderboardSlug = route.params.slug as string
+
+const leaderboardDetail = await useGetLeaderboardBySlug(leaderboardSlug)
```
Comment (TheTedder) [inline pages/game/[slug].vue:9]: Use destructuring assignment here.

**Review by TheTedder — APPROVED** [review]

**Comment by Eein** [comment]
had to rebind error so tests can pass due to:

`[Uncaught TypeError: 'set' on proxy: trap returned falsish for property 'error'`

File: pages/game/[slug].vue [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+const {
+  loading,
+  error: leaderboardError,
+  data,
+} = await useGetLeaderboardBySlug(slug as string)
+
+if (leaderboardError?.status === 404) {
+  throw createError({
+    statusCode: 404,
+    statusMessage: 'Page Not Found',
```
Comment (zysim) [inline pages/game/[slug].vue:19]: Maybe this should say "Board Not Found" instead?

Comment (erunks) [reply pages/game/[slug].vue:19]: I can get behind this

**Pull request merged**